doc/howto/dqlite_troubleshoot.md

```diff
@@ -0,0 +1,85 @@
+contents of the dqlite data directory for the affected node. This is the
+directory where the local instance of dqlite stores all its data. If you
+installed the LXD snap, you can find this directory at
+`/var/snap/lxd/common/lxd/database/global`.
+
+The data directory contains several types of file. The most important types are:
+
+- Closed segments: these have filenames like
+  `0000000000056436-0000000000056501`. The two numbers are the *start index*
+   and *end index*. Both indices are inclusive.
+- Open segments: these have filenames like `open-1`
+- Snapshot files: these have names like `snapshot-1-59392-27900`. The first
+  number is the *snapshot index*.
+- Snapshot metadata files: these have names like `snapshot-1-59392-27900.meta`
+  and are paired with snapshot files.
+
+## Spotting weirdness
+
+When looking at the contents of the data directory, watch for the following
+symptoms:
+
+1. Closed segments whose index ranges overlap (remember that these ranges are inclusive)
+2. A closed segment with end index X where the next closed segment has start index greater than X + 1
+3. A snapshot file with snapshot index X where the next closed segment has start index greater than X + 1
+4. A snapshot file whose size is less than the size of a previous (lower-numbered) snapshot
+
+When scanning for these symptoms, start with the most recent snapshots and
+closed segments (those with the highest indices) since the problem is more
+likely to be there.
+
+## Specific error messages
+
+- `closed segment [...] is past last snapshot [...]`: this indicates that you have symptom (3) above, possibly combined with symptom (1)
+- `load closed segment [...]: entries count in preamble is zero`: this indicates that the mentioned segment is corrupt
+
+## Interventions
+
+**Important:** Before taking any of the actions below, back up the entire
+dqlite data directory, so you don't lose data in case something goes wrong.
+
+Here are some actions you can take in response to specific dqlite errors. They
+are aimed at restoring the invariants described above without destroying any
+data. They are not guaranteed to work in any specific case.
+
+- If you have overlapping closed segments (symptom 1), try deleting some of
+  them to remove the overlap, without creating gaps in the sequence of indices
+  or removing any index that was previously represented
+- If the snapshot file with the highest index is unexpectedly small (symptom
+  4), and there are still closed segments covering all the indices up to and
+  including this snapshot's index, delete the snapshot and its corresponding
+  metadata file
+- If the last (highest-numbered) closed segment is corrupt, try deleting it.
+  (Deleting closed segments before the last one will create a gap and generally
+  prevent dqlite from starting.)
+
+## Get help
+
+If the tips above don't help with your situation, you can always post on the
+LXD support forum. Please mention dqlite in your post, tag @colemiller and
```

Comment:
```suggestion
LXD support forum. Please mention dqlite in your post, tag `@colemiller` and
```

Reply:
Not sure I like mentioning specific user names in the docs ... do we have a group we could mention instead?

Reply:
agreed, lets just reference the support forum